Std/Tactic/PrintPrefix.lean

```diff
@@ -0,0 +1,48 @@
+/-
+Copyright (c) 2021 Shing Tak Lam. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shing Tak Lam, Daniel Selsam, Mario Carneiro
+-/
+import Std.Lean.Util.EnvSearch
+import Lean.Elab.Command
+
+open Lean Meta Elab Command
+
+namespace Lean
+
+
+namespace Elab.Command
+
+
+private def appendMatchingConstants (msg : String)
+    (ϕ : ConstantInfo → MetaM Bool) (opts : EnvironmentSearchOptions := {}) : MetaM String := do
+  let cinfos ← getMatchingConstants ϕ opts
+  let cinfos := cinfos.qsort fun p q => p.name.lt q.name
+  let mut msg := msg
+  for cinfo in cinfos do
+    msg := msg ++ s!"{cinfo.name} : {← Meta.ppExpr cinfo.type}\n"
+  pure msg
+
+/--
+Command for #print prefix
+-/
+syntax (name := printPrefix) "#print prefix " ident : command
```

Comment:
My personal version of used `"#print" &"prefix"` rather than `"#print prefix"`. I'm not sure my version is better but I am interested in learning more about the difference. (It's been so long I don't remember why I did that!)

Reply:
It looks like this dates back at least to when it was [checked into Mathlib](https://github.com/leanprover-community/mathlib4/commit/81e736ed58fe4f5ef6bba68f4aeb84916d7ea093) by @digama0

Reply:
``"#print " "prefix "` is better if the intent is to allow more or less whitespace between the tokens. One probably needs to also add a `colGt` or something, or else unfinished `#print` statements cannot be differentiated from `#print prefix` where the `prefix` just happens to be the next command.

Reply:
Interesting, I needed to learn this.  It seems like I should allow whitespace, but I noticed that i also need `withPosition`

Consider these two examples:

```
#print
  prefix
  Nat

#print
prefix
Nat
```

I think I want to parse the first, but not the second.  However the line below parses both:

```
syntax (name := printPrefix) "#print " (colGt "prefix ") (colGt ident) : command
```

This works though:

```
syntax (name := printPrefix) withPosition("#print " (colGt "prefix ") (colGt ident)) : command
```

Any feedback?  I was surprised I needed an explicit `withPosition` since other similar examples didn't seem to use it.

Reply:
Yes, you will need a `withPosition` because commands are parsed at the top level and this doesn't have a position set. (Tactics, by contrast, appear somewhere after a `by` and this sets the position.)